freedomofpress/securedrop-workstation
# Support automatic attachment of export devices

Adds udev rules and scripts to `sys-usb`. Modifies the qubes.USB RPC policy and adds qubes.USBAttach.

## Testing

- Check out this branch in your development Qube.
- In your cloned repo in `dom0`:
  - `make clone`
  - `make clean && make all`
  - Check that `/etc/qubes-rpc/policy/qubes.USB` contains:
    ```
    ### BEGIN securedrop-workstation ###
    sd-export-usb sys-usb allow
    @anyvm @tag:sd-workstation deny
    ### END securedrop-workstation ###
    @anyvm @anyvm ask
    ```
  - Check that `/etc/qubes-rpc/policy/qubes.USBAttach` contains:
    ```
    ### BEGIN securedrop-workstation ###
    sys-usb sd-export-usb allow,user=root
    @anyvm @tag:sd-workstation deny
    ### END securedrop-workstation ###
    @anyvm @anyvm ask
    ```
- In `sys-usb`, confirm the existence of:
  - `/etc/udev/rules.d/99-sd-export-usb.rules`
  - `/usr/local/bin/sd-attach-export-disk`
  - `/usr/local/bin/sd-attach-export-printer`

Plug in a valid LUKS export stick. It should be automatically attached to `sd-export-usb`, and an export from the client or manually via `qvm-open-in-vm sd-export-usb "filename"` should succeed.
